### PR TITLE
feat(ToggleSwitch): Google Material style updates

### PR DESCRIPTION
**Pull request by mdodgelooker**
- Updated the styles and sub-components of `ToggleSwitch`
- Added the ripple effect (toggle "ripple" in the storybook addon controls to see the animation)
- Realized the `readOnly` prop wasn't doing anything, fixed that
- Added `critical` / `criticalAccent` color for error state – I'm not sure if this is appropriate though?

- [x] 🖼 Image Snapshot coverage
- [x] 👾 Browsers tested
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [ ] IE11 (neither the playground nor the storybook iframe seem to be working in IE11 anymore)

0b04728f-cbb5-411f-9b0f-baab653f031c
d08a6d10-f5b6-424b-8138-ff8aa2a02303

### CODE REVIEW

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 1]
```diff
@@ -25,71 +25,105 @@
...
-        {disabled && <DisabledKnob />}
-      </div>
-    )
-  }
-)
-
-ToggleSwitchLayout.displayName = 'ToggleSwitchLayout'
-
-export const ToggleSwitch = styled(ToggleSwitchLayout)`
+export const ToggleSwitch = styled(
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx:54]: "Hide whitespace changes" helps a little with this diff.

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 2]
```diff
@@ -25,71 +25,105 @@
...
+            ref={ref}
+            {...pickInputProps(props)}
+          />
+          <Track on={on} validationType={validationType} />
+          <Handle
+            on={on}
+            validationType={validationType}
+            className={rippleClassName}
+            style={style}
+          ></Handle>
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx:107]: Stole "Track" and "Handle" from the design spec. 🙂

File: packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx [hunk 3]
```diff
@@ -25,71 +25,105 @@
...
+  )
+)`
   ${reset}
   ${space}
 
-  height: 1.25rem;
+  align-items: center;
+  display: flex;
+  height: ${({ theme }) => theme.space.u6};
+  justify-content: center;
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/ToggleSwitch/ToggleSwitch.tsx:119]: Use flex to get the `Track` in the middle.

File: packages/components/src/Form/Inputs/ToggleSwitch/Handle.tsx [hunk 4]
```diff
@@ -24,54 +24,43 @@
...
-  transform: ${({ on }) => (on ? `translateX(0.9375rem)` : undefined)};
-  transition: ${({ theme }) => theme.transitions.moderate}ms;
-  width: 1rem;
-`
-
-interface KnobContainerProps extends KnobProps {
-  className?: string
+const getColor = ({ on, validationType }: SwitchElementProps): keyof Colors => {
+  if (on) {
+    if (validationType === 'error') return 'critical'
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply packages/components/src/Form/Inputs/ToggleSwitch/Handle.tsx:35]: I kinda like it - not sure what the use case would be but I don't see a reason _not_ to include it. 🤷  (@jhardy ?)

**Review by lukelooker — DISMISSED** [review]
Looks great!

File: packages/components/src/Form/Inputs/ToggleSwitch/Handle.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhardy) [reply packages/components/src/Form/Inputs/ToggleSwitch/Handle.tsx:35]: I am not sure the use case but its kind of nice that is aligns with the other controls. I don't see a downside to including it.

**Comment by mdodgelooker** [comment]
Sped up the animation to `theme.transitions.rapid` which is 100ms to more closely match what [Material has](https://material-components.github.io/material-web/demos/switch/), 75ms. (Per @jhardy we may have to adjust our transitions at some point.)

**Review by jhardy — APPROVED** [review]
LGTM

**Pull request merged**